cpp/src/lists/contains.cu

```diff
@@ -46,10 +52,52 @@ auto get_search_keys_device_iterable_view(cudf::scalar const& search_key, rmm::c
   return &search_key;
 }
 
+/**
+ * @brief Enum to indicate whether the `search_key` scalar/column contains nulls.
+ */
+enum search_key_nulls : bool { NO_NULLS = false, HAS_NULLS = true };
```

Comment:
Why is this better than just a `bool`?

Reply:
I hoped that it would make the [call site](https://github.com/rapidsai/cudf/pull/9510/files#diff-d66d386d623323d882af299672d1a96abf1c8ade91dcf2cf0ac529e3b45fb329R265) marginally more legible:
```c++
lookup_functor<search_key_nulls::NO_NULLS>{}
```
vs.
```c++
lookup_functor<false>{}
```
It might be moot here, since it's used in just one place. I can change it back, if that's preferable.

Reply:
I have switched this back to use bools.
I wanted to move this to use `cudf::nullate`, but I can't yet. (Looks like `make_pair_rep_iterator()` needs to support `nullate` first.) I'll take this up in a follow-on.

Reply:
Filed #9869 to add `nullate` support in `make_pair_rep_iterator`, etc. Will tackle the `nullate` switch there.

Reply:
What is "nullate"? I googled it but can't find the word.

Reply:
For the sake of closure, I should summarize here what was discussed offline.

The [`nullate`](https://github.com/rapidsai/cudf/blob/branch-22.02/cpp/include/cudf/column/column_device_view.cuh#L53) device is the mechanism introduced to replace the explicit `has_nulls` bools that flagged conditional handling for columns that contain null rows.

The `lookup_functor` is a prime candidate to use `nullate`:
```c++
lookup_functor{nullate::NO} // instead of lookup_functor<NO_NULLS>{}
```
It requires rewiring `make_pair_rep_iterator()`, so that change is postponed for now.